MaterialDesignThemes.Wpf/Snackbar.cs

```diff
@@ -31,7 +31,16 @@ static Snackbar()
         }
 
         public static readonly DependencyProperty MessageProperty = DependencyProperty.Register(
-            nameof(Message), typeof(SnackbarMessage), typeof(Snackbar), new PropertyMetadata(default(SnackbarMessage)));
+            nameof(Message), typeof(SnackbarMessage), typeof(Snackbar), new PropertyMetadata(default(SnackbarMessage), MessagePropertyChangedCallback));
+
+        private static void MessagePropertyChangedCallback(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs dependencyPropertyChangedEventArgs)
+        {
+            var snackbar = (Snackbar)dependencyObject;
+            //this prevents missing resource warnings after the message is removed from the Snackbar
+            //see https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/issues/2040
+            if (snackbar.Message != null)
+                snackbar.Message.Resources = SnackbarMessage.defaultResources;
```

Comment:
"UNDO" has disappeared. It should be like this.

```csharp
            if (dependencyPropertyChangedEventArgs.OldValue is SnackbarMessage oldMessage)
                oldMessage.Resources = SnackbarMessage.defaultResources;
```

Since we can reuse `SnackbarMessage` like this, "UNDO" disappears with the current approach.

```csharp
      private async void Button1_Click (object sender, RoutedEventArgs e)
        {
            var message1 = new SnackbarMessage { Content = "Hello world1", ActionContent = "UNDO" };
            var message2 = new SnackbarMessage { Content = "Hello world2", ActionContent = "UNDO" };

            MySnackbar.Message = message1;
            MySnackbar.IsActive = true;
            await Task.Delay (1000);

            MySnackbar.Message = message2;
            await Task.Delay (1000);

            MySnackbar.Message = message1;
            await Task.Delay (1000);
            MySnackbar.IsActive = false;
        }
```

Therefore, this may be better.

Snackbar:
```csharp
        private static void MessagePropertyChangedCallback(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs dependencyPropertyChangedEventArgs)
        {
            (dependencyPropertyChangedEventArgs.OldValue as SnackbarMessage)?.Clear();
            (dependencyPropertyChangedEventArgs.NewValue as SnackbarMessage)?.Restore();
        }
```

SnackbarMessage:
```csharp
        private ResourceDictionary _resources;
        public new ResourceDictionary Resources
        {
            get => _resources;
            set
            {
                _resources = value;
                if (base.Resources != defaultResources)
                    base.Resources = value;
            }
        }

        public SnackbarMessage() => _resources = base.Resources;
        internal void Clear() => base.Resources = defaultResources;
        internal void Restore() => base.Resources = _resources;
```

Reply:
Yes, you are right.  Great catch.

In the past two days, I have been trying to find a better solution.  To that end, I thought it would be helpful if I could reproduce this "Resource not found" warning using a minimal example (i.e. one that doesn't depend on MDinXAML).  I have tried, but I haven't been able to figure out how to do that.

@bombipappoo, do you know how to do this?  Said another way...do you know how to create an example like I did in https://github.com/dotnet/wpf/issues/2956#issuecomment-681189922 but one that is actually minimal (...one that doesn't have any dependencies like MDinXAML)?